shared/src/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,256 @@
+     * left and right nav items.
+     */
+    navProps?: HTMLProps<HTMLElement> | null;
+    /**
+     * Additional props to be applied to the nav element that wraps the
+     * navigation items on the right hand side.
+     */
+    rightNavProps?: HTMLProps<HTMLUListElement> | null;
+    /**
+     * Props to pass to the SearchBox component. When these props are provided the
+     * search box will appear.
+     */
+    searchProps?: SubComponentProps<SearchBoxProps> | null;
+    /**
+     * The header theme. When this is not provided the header will use the default
+     * theme defined in the Vanilla settings.
+     */
+    theme?: Theme | null;
+  },
+  HTMLProps<HTMLElement>
+>;
+
+/**
+ * Narrow the type of the nav item to a NavMenu.
+ */
+const isMenu = (item: NavItem): item is NavMenu => "items" in item;
+
+/**
+ * Narrow the type of the logo prop to LogoProps.
+ */
+const isLogoProps = (logo: Props["logo"]): logo is LogoProps =>
+  !isValidElement(logo);
+
+/**
+ * Generate the JSX for a set of nav items. This will map the items to menus,
+ * links or generated components.
+ * @param items The nav items to map to elements.
+ * @param closeMobileMenu A function to close the mobile menu.
+ * @param generateLink The optional function used to generate link components.
+ * @returns A list of navigation item elements.
+ */
+const generateItems = (
+  items: NavItem[],
+  closeMobileMenu: () => void,
+  generateLink?: GenerateLink | null
+) =>
+  items.map((item, i) =>
+    isMenu(item) ? (
+      <NavigationMenu {...item} generateLink={generateLink} key={i} />
+    ) : (
+      <li
+        className={classNames("p-navigation__item", {
+          "is-selected": item.isSelected,
+        })}
+        key={i}
+      >
+        <NavigationLink
+          {...item}
+          onClick={(evt) => {
+            item.onClick?.(evt);
+            closeMobileMenu();
+          }}
+          generateLink={generateLink}
+          className={classNames("p-navigation__link", item.className)}
+        />
+      </li>
+    )
+  );
+
+const Navigation = ({
+  generateLink,
+  items,
+  itemsRight,
+  leftNavProps,
+  logo,
+  navProps,
+  rightNavProps,
+  searchProps,
+  theme,
+  ...headerProps
+}: Props): JSX.Element => {
+  const [mobileMenuOpen, setMobileMenuOpen] = useState(false);
+  const [searchOpen, setSearchOpen] = useState(false);
+  // Display the search box if the props have been provided.
+  const hasSearch = !!searchProps;
+  // Close the mobile menu when the search box is opened.
+  const toggleSearch = () => {
+    setSearchOpen(!searchOpen);
+    setMobileMenuOpen(false);
+  };
+  // Close the search box when the mobile menu is opened.
+  const toggleMobileMenu = () => {
+    setMobileMenuOpen(!mobileMenuOpen);
+    setSearchOpen(false);
+  };
+  const closeMobileMenu = () => {
+    if (mobileMenuOpen) {
+      setMobileMenuOpen(false);
+    }
+  };
+  return (
+    <header
+      id="navigation"
```

Comment:
This could just be part of `headerProps`, no?

Reply:
Yeah, thanks.